utils/crates-io/src/publisher.rs

```diff
@@ -61,53 +61,85 @@ impl Publisher {
     /// 1. Replace git dependencies to crates-io dependencies.
     /// 2. Rename version of all local packages
     /// 3. Patch dependencies if needed
-    pub fn build(mut self, verify: bool, version: Option<String>) -> Result<Self> {
-        let mut index = self
-            .index
-            .iter()
-            .map(|(k, &v)| (k.clone(), v))
-            .collect::<Vec<_>>();
-
-        index.sort_by_key(|(_, v)| *v);
+    pub fn build(mut self, verify: bool, mut version: Option<String>) -> Result<Self> {
+        // Do not use version in simulation due to specificities of `cargo_http_registry`
+        if self.simulator.is_some() {
+            version.take();
+        }
 
         let mut workspace = Workspace::lookup(version)?;
         let version = workspace.version()?;
 
-        for (name, _) in &index {
+        for name in self.index.iter() {
             let Some(pkg) = self.metadata.packages.iter().find(|pkg| pkg.name == *name) else {
+                println!("Package {name}@{version} not found in cargo metadata!");
                 continue;
             };
 
-            if verify && crate::verify(name, &version)? {
-                println!("Package {}@{} already published .", &name, &version);
+            // TODO: remove this check
```

Comment:
^ try it with one package on local machine: `cargo +stable run --release -p crates-io publish --simulate`